app/scripts/services/membership/membership.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .factory('MembershipService', function($filter) {
+
+    var annotation = $filter('annotation');
+
+    var isLastRole = function(userName, roleBindings) {
+      return _.filter(
+              roleBindings,
+              function(binding) {
+                return _.some(binding.subjects, { name: userName });
+              }).length === 1;
+    };
+
+    // internal helper to disambiguate objects with same name, etc:
+    //  - Role view
+    //  - ClusterRole view
+    // arbitrary key creation via any number of provided arguments
+    // examples:
+    //  - uniqueKey(subject.kind, subject.name) = user-foo
+    //  - uniqueKey(role.namespace, role.name) = proj-1-view
```

Comment:
Thought about that.  It has been easier to fish around in the data with a human readable name during dev, but at this point the UID prob makes sense.